src/Kernel/CompiledMethod.class.st

```diff
@@ -437,9 +437,11 @@ CompiledMethod >> isExplicitlyRequired: marker [
 ]
 
 { #category : #testing }
-CompiledMethod >> isExtensionMethod [
+CompiledMethod >> isExtension [
 
-	^ self protocol isExtensionProtocol
+	^ self protocol
+		  ifNil: [ false ]
+		  ifNotNil: [ :protocol | protocol isExtensionProtocol ]
```

Comment:
It is shame that packaging is implemented in terms of a special protocol names. It should be a real thing independent from protocols. The original implementation was about this in mind. Now this PR pushes it back.
Imagine that one day the packaging system will be improved - you will need to change isExtension again. 

The "package API" is in place to not rely on protocols even if underhood it is still the foundation. If there are bugs they should be fixed. But any new place where extension protocols are exposed will complicate the future improvement to decouple protocols from packages.

Reply:
Hi Denis,
In the long run I also want to remove the fact that extensions rely on protocols. But currently, RPackage also rely on that everywhere.

My goal is to clean RPackage to make it way simpler because currently I’ve never seen code so brittle as RPackage and I think it would be a considerable effort to not rely on protocol. Once RPackage will be integrated in the MM and the complexity removed, I think it will be easier to remove the usage of extension protocol. But currently, to remove the complexity of RPackage we need to speed up the meta model. And lot of speed problems comes from RPackage itself and the way classes and methods are using it.

I can add a comment explaining that this is temporary and that in the future we want to remove the fact that protocols are used for extensions. But asking for packages to a class or method currently is too long because of the implementation and doing big changes to RPackage is just impossible due to its state.  

Would the comment I proposed to add be enough?

Reply:
I am happy that you have plan :)

Reply:
I added a comment with:
- A description of the method
- A explanation about the temporary usage of protocol and how it should be done in the future when RPackage will be faster
- Executable examples

I know it's not nice to use the protocol but it's too hard to clean RPackage and update the way extension methods work at the same time :(